File: c#/Bouchonnois.Tests/Builders/ChasseurBuilder.cs
using Bouchonnois.Domain;

namespace Bouchonnois.Tests.Builders
{
    public class ChasseurBuilder
    {
        private string? _nom;
        private int _ballesRestantes;
        private int _nbGalinettes;

        public ChasseurBuilder(string nom) => _nom = nom;

        private ChasseurBuilder(string nom, int ballesRestantes)
        {
            _nom = nom;
            _ballesRestantes = ballesRestantes;
        }

        public ChasseurBuilder SansBalles()
        {
            _ballesRestantes = 0;
            return this;
        }

        public ChasseurBuilder AyantTué(int nbGalinettes)
        {
            _nbGalinettes = nbGalinettes;
            return this;
        }

        public static ChasseurBuilder LeChasseur(string nom) => new(nom);
        public static ChasseurBuilder Dédé() => new ChasseurBuilder("Dédé", 20);
        public static ChasseurBuilder Bernard() => new ChasseurBuilder("Bernard", 8);
        public static ChasseurBuilder Robert() => new ChasseurBuilder("Robert", 12);

        public Chasseur Build() => new(_nom!) {BallesRestantes = _ballesRestantes, NbGalinettes = _nbGalinettes};
    }
}

File: java/src/test/java/bouchonnois/tests/builders/ChasseurBuilder.java
package bouchonnois.tests.builders;

import bouchonnois.domain.Chasseur;

public class ChasseurBuilder {
    private final String nom;
    private int ballesRestantes;
    private int nbGalinettes;

    public ChasseurBuilder(String nom) {
        this.nom = nom;
    }

    private ChasseurBuilder(String nom, int ballesRestantes) {
        this.nom = nom;
        this.ballesRestantes = ballesRestantes;
    }

    public static ChasseurBuilder leChasseur(String nom) {
        return new ChasseurBuilder(nom);
    }

    public static ChasseurBuilder dédé() {
        return new ChasseurBuilder("Dédé", 20);
    }

    public static ChasseurBuilder bernard() {
        return new ChasseurBuilder("Bernard", 8);
    }

    public static ChasseurBuilder robert() {
        return new ChasseurBuilder("Robert", 12);
    }

    public ChasseurBuilder sansBalles() {
        this.ballesRestantes = 0;
        return this;
    }

    public ChasseurBuilder ayantTué(int nbGalinettes) {
        this.nbGalinettes = nbGalinettes;
        return this;
    }

    public Chasseur build() {
        return new Chasseur() {{
            setNom(nom);
            setBallesRestantes(ballesRestantes);
            setNbGalinettes(nbGalinettes);
        }};
    }
}
